swift/modules/SwiftUtilities/Sources/SwiftUtilities/String+uniqueName.swift

```diff
@@ -0,0 +1,37 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+import Foundation
+
+public extension String {
+    /// Returns a unique filename for use in testing. The filename
+    /// is just a UUID, optionally with a period and an extension
+    /// added. The returned name is invalid for S3 use if isValid is
+    /// true.
```

Comment:
/// Returns a unique file name for use in testing. The file name
    /// is a UUID. You can optionally add a period and an extension
    /// to it. If isValid is true, the returned name isn't valid for use 
    /// with Amazon Simple Storage Service (Amazon S3).